internal/nozzle/nozzle.go

```diff
@@ -41,13 +46,39 @@ type Nozzle struct {
 	totalMessagesReceived uint64            // modified by workers, read by main thread
 	slowConsumerAlert     uint64            // modified by workers, read by main thread
 	totalMetricsSent      uint64
+	stopConsumer          func()
 }
 
 // AuthTokenFetcher is an interface for fetching an auth token from uaa
 type AuthTokenFetcher interface {
 	FetchAuthToken() string
 }
 
+type rlpGatewayClientDoer struct {
```

Comment:
Should we have this logic as part of our clients package instead. Do we need a wrapper object around the loggregator?

Reply:
Yeah, I think that makes sense. I'll try to do some refactoring here.